src/components/HomePage.jsx

```diff
@@ -1,7 +1,58 @@
+import { useEffect, useState } from 'react';
+import fetchArticles from './api/fetchArticles';
```

Comment:
I would recommend breaking your api down to a base url and calls per action e.g. getArticles, getSingleArticle, etc so you can add all your api calls to one file to avoid too many files, but you could do it by the type of data if you want to